### PR TITLE
Refator Egg Dropping Implementation

### PR DESCRIPTION
**Pull request by TruongNhanNguyen**
## Checklist:

- [x] I ran bellow commands using the latest version of **rust nightly**.
- [x] I ran `cargo clippy --all -- -D warnings` just before my last commit and fixed any issue that was found.
- [x] I ran `cargo fmt` just before my last commit.
- [x] I ran `cargo test` just before my last commit and all tests passed.
- [x] I added my algorithm to the corresponding `mod.rs` file within its own folder, and in any parent folder(s).
- [x] I added my algorithm to `DIRECTORY.md` with the correct link.
- [x] I checked `COUNTRIBUTING.md` and my code follows its guidelines.

### CODE REVIEW

**Comment by TruongNhanNguyen** [comment]
Can you have a look, @vil02.

**Comment by TruongNhanNguyen** [comment]
Please have a look, @vil02.

File: src/dynamic_programming/egg_dropping.rs [hunk 1]
```diff
@@ -1,91 +1,84 @@
...
-        egg_drops[1][j] = j as u32;
+/// # Arguments
+///
+/// * `eggs` - The number of eggs available.
+/// * `floors` - The number of floors in the building.
+///
+/// # Returns
+///
+/// * `Some(usize)` - The minimum number of drops required if the number of eggs is greater than 0.
+/// * `None` - If the number of eggs is less than or equal to 0.
```
Comment (vil02) [inline src/dynamic_programming/egg_dropping.rs:15]:
```suggestion
/// * `None` - If the number of eggs is 0.
```
`eggs` is of the type `usize`.

**Review by vil02 — CHANGES_REQUESTED** [review]

File: src/dynamic_programming/egg_dropping.rs (hunk 1 above)
Comment (TruongNhanNguyen) [reply src/dynamic_programming/egg_dropping.rs:15]: Thanks.

File: src/dynamic_programming/egg_dropping.rs [hunk 2]
```diff
@@ -1,91 +1,84 @@
...
-        egg_drops[1][j] = j as u32;
+/// # Arguments
+///
+/// * `eggs` - The number of eggs available.
+/// * `floors` - The number of floors in the building.
+///
+/// # Returns
+///
+/// * `Some(usize)` - The minimum number of drops required if the number of eggs is greater than 0.
+/// * `None` - If the number of eggs 0.
```
Comment (vil02) [inline src/dynamic_programming/egg_dropping.rs:15]:
```suggestion
/// * `None` - If the number of eggs is 0.
```

**Pull request merged**